src/libstore/dummy-store.cc

```diff
@@ -56,6 +56,11 @@ struct DummyStore : public virtual DummyStoreConfig, public virtual Store
     void queryRealisationUncached(const DrvOutput &,
         Callback<std::shared_ptr<const Realisation>> callback) noexcept override
     { callback(nullptr); }
+
+    bool checkTrustedUser() override
+    {
+        throw Error("You cannot query trusted-user on a dummy store");
```

Comment:
On the advice of @roberth I decided not to do this. If I was going to do this, I'd just make the default implementation in the `virtual` function in `store-api.cc` say "This store type is not supported" for example.

Reply:
A default implementation was what I had in mind, yes. My bad for being unclear.

Reply:
@roberth seems not to like default implementations, and I also think it's a good idea to avoid default implementations, generally. It's a small nitpick and something that seems to vary between programmers.